provider/file/file.go

```diff
@@ -18,68 +22,135 @@ var _ provider.Provider = (*Provider)(nil)
 // Provider holds configurations of the provider.
 type Provider struct {
 	provider.BaseProvider `mapstructure:",squash"`
+	Directory             string `description:"Load configuration from one or more .toml files in a directory"`
 }
 
 // Provide allows the file provider to provide configurations to traefik
 // using the given configuration channel.
 func (p *Provider) Provide(configurationChan chan<- types.ConfigMessage, pool *safe.Pool, constraints types.Constraints) error {
-	watcher, err := fsnotify.NewWatcher()
+	var configuration *types.Configuration
+	var err error
+	var watchDir string
+
+	if p.Directory != "" {
+		watchDir = p.Directory
+		configuration, err = loadFileConfigFromDirectory(p.Directory)
+	} else {
```

Comment:
Ideally, yes, we'd notify if both a single file and directory are specified.  However, file is always specified, as it's then name of the main Traefik config file, even if the `filename` isn't set in the `[file]` config section.  I didn't see a straightforward way to address this.